agent/engine/docker_task_engine.go

```diff
@@ -1336,25 +1340,55 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 
 		containerCredSpec, err := container.GetCredentialSpec()
 		if err == nil && containerCredSpec != "" {
-			// CredentialSpec mapping: input := credentialspec:file://test.json, output := credentialspec=file://test.json
+			// on indows CredentialSpec mapping: input := credentialspec:file://test.json, output := credentialspec=file://test.json
+			// on linux CredentialSpec mapping: input := ssm/asm arn, output := /var/credentials-fetcher/krbdir/123456/ccname_webapp01_xyz
 			desiredCredSpecInjection, err := credSpecResource.GetTargetMapping(containerCredSpec)
-			if err != nil || desiredCredSpecInjection == "" {
-				missingErr := &apierrors.DockerClientConfigError{Msg: "unable to fetch valid credentialspec mapping"}
-				return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(missingErr)}
-			}
+			if runtime.GOOS == "windows" {
```

Comment:
can we separate this behavior by build flags rather than runtime.GOOS ?

Reply:
> Is it possible to break this PR down into multiple short ones? 280 files changed is too much to review in 1 PR.

Most of the changes are in vendor repo because of gRPC dependency

Reply:
using osType instead